app/scripts/metamask-controller.js

```diff
@@ -1507,10 +1506,39 @@ export default class MetamaskController extends EventEmitter {
       signatureController: this.signatureController,
       platform: this.platform,
       extension: this.extension,
-      trackTransactionEvents: handleMMITransactionUpdate.bind(
-        null,
-        transactionMetricsRequest,
-      ),
+      hooks: {
+        getTransactions: this.txController.getTransactions.bind(
+          this.txController,
+        ),
+        updateTransactionHash: this.txController.setTxHash.bind(
+          this.txController,
+        ),
+        trackTransactionEvents: handleMMITransactionUpdate.bind(
+          null,
+          transactionMetricsRequest,
+        ),
+        txStateManager: {
+          getTransactions: this.txController.getTransactions.bind(
+            this.txController,
+          ),
+          setTxStatusSigned:
+            this.txController.txStateManager.setTxStatusSigned.bind(
```

Comment:
I was thinking of only passing what is required to ensure that until the migration no new method can be used from `txStateManager`.
The idea is to cover all the `txStateManager` methods required here with the `state` of the core `TransactionController` and ` updateTransaction` but as you pointed out we eventually will pull down the `updateTransaction`, so I included a specific method in core to handle the transaction update [here](https://github.com/MetaMask/core/pull/2018).

Reply:
For audit sake, rather than add getters to the controller such as `getTransactions`, the more flexible pattern that has become standard is just reading the state directly and extracting whatever is needed.

In this case, in future, we could bind to `() => this.txController.state.transactions` for example.